docs/guides/adapters.md

```diff
@@ -0,0 +1,115 @@
+- [ ] add option in [adapters/index.js](adapters/index.js)
+
+In the adapters [adapters/index.js](adapters/index.js) in the  `sttJsonAdapter` function switch statement add a new `case` with the new STT service type eg `speechmatics`
+
+<!-- TODO: modify import path if module is moved/refactored -->
+```js
+import speechmaticsToDraft from './speechmatics/index';
+
+...
+
+case 'speechmatics':
+      blocks = speechmaticsToDraft(transcriptData);
+      return { blocks, entityMap: createEntityMap(blocks) };
+```
+
+- [ ] add an adapter function.
+
+as shown in the example you'd also need to add a function with the stt provider name +`ToDraft` eg `speechmaticsToDraft`that takes in the transcript data.
+
+- [ ] create a function to convert the STT data structure into draftJs blocks and entityRanges.
+
+You can see examples from `bbc-kaldi` and `autoEdit2` adapters.
+
+In pseudocode it's reccomended to follow this approach:
+
+1. Expose one function call that takes in the stt json data
+2. have a helper function `groupWordsInParagraphs` that as the name suggests groups words list from the STT provider transcript based on punctuation. and returns an array of words objects.
+
+The underlying details for this will vary depending on how the STT json of the provider present the data, and how the attributes are named etc..
+
+3. iterate over the paragraphs to create draftJS content blocks (see `bbc-kaldi` and `autoEdit2` example).
+
+```js
+wordsByParagraphs.forEach((paragraph, i) => {
+    const draftJsContentBlockParagraph = {
+      text: paragraph.text.join(' '),
+      type: 'paragraph',
+      data: {
+        speaker: `TBC ${ i }`,
+        words: paragraph.words, //
+        start: paragraph.words[0].start//
+      },
+      // the entities as ranges are each word in the space-joined text,
+      // so it needs to be compute for each the offset from the beginning of the paragraph and the length
+      entityRanges: generateEntitiesRanges(paragraph.words, 'text'), // wordAttributeName
+    };
+    // console.log(JSON.stringify(draftJsContentBlockParagraph,null,2))
+    results.push(draftJsContentBlockParagraph);
+  });
+
+```
+
+4. and use the helper function `generateEntitiesRanges` to  add the `entityRanges` to each block. - see above
+
+5. If you have speaker diarization info you can also add this to the block info - _optional_
+
+
+## tests
+
+This project uses jest. and TravisCI. It is recommended to write some basic tests at a minimum so that you can see at a glance if the adapter is working as expected (see `bbc-kaldi` and `autoEdit2` example). 
+
+_If you don't have much experience with automated testing don't let this put you off tho, feel free to raise it as an issue and we can help out._
+
+**top tip**: when preparing your expected results for the test, you might need to allow jest to be less specific about the key attributes, these are randomly generated, and therefore cannot be tested in a deterministic way.
```

Comment:
I get that this is useful here - but it's still 3 lines which don't fully clear up _what_ is randomly generated etc. I think there's too many words here, and they don't really get to the point.

Reply:
ok, suggestions on re-phrasing welcome 🙂